packages/rocketchat-api/server/v1/chat.js

```diff
@@ -255,3 +255,23 @@ RocketChat.API.v1.addRoute('chat.update', { authRequired: true }, {
 		});
 	}
 });
+
+RocketChat.API.v1.addRoute('chat.react', { authRequired: true }, {
+	post() {
+		if (!this.bodyParams.messageId || !this.bodyParams.messageId.trim()) {
+			throw new Meteor.Error('error-messageid-param-not-provided', 'The required "messageId" param is missing.');
+		}
+
+		const msg = RocketChat.models.Messages.findOneById(this.bodyParams.messageId);
+
+		if (!msg) {
```

Comment:
You can move this verification to see if have or not message to the setReactions method, cuz already has one call to db for retrieve message, and you can save a call.

Reply:
I'm going to have to disagree with this. That is outside of the scope of this pull request. If we change how the `setReaction` works then we are going to have to change how the other places which calls that method handles the result. The `setReaction` method doesn't return errors when the message doesn't exist and it only returns `false` so I think we should keep this error here for other developer's sake.

Reply:
Ok @graywolf336, I had not througt about it. You're right.